tests/components/updater/test_init.py

```diff
@@ -44,38 +45,76 @@ def mock_get_uuid():
         yield mock
 
 
+@pytest.fixture
+def mock_dispatcher_connect():
+    """Fixture to mock dispatcher_connect."""
+    with patch("homeassistant.helpers.dispatcher.async_dispatcher_connect") as mock:
+        yield mock
+
+
 @asyncio.coroutine
-def test_new_version_shows_entity_after_hour(
-    hass, mock_get_uuid, mock_get_newest_version
-):
+def test_new_version_shows_entity_startup(hass, mock_get_uuid, mock_get_newest_version):
+    """Test if new entity is created if it is unavailable at first."""
+    mock_get_uuid.return_value = MOCK_HUUID
+    mock_get_newest_version.return_value = mock_coro((NEW_VERSION, RELEASE_NOTES))
+
+    res = yield from async_setup_component(hass, updater.DOMAIN, {updater.DOMAIN: {}})
+    assert res, "Updater failed to set up"
+
+    yield from hass.async_block_till_done()
+
+    assert hass.states.is_state("binary_sensor.updater", "unavailable")
+    assert "newest_version" not in hass.states.get("binary_sensor.updater").attributes
+    assert "release_notes" not in hass.states.get("binary_sensor.updater").attributes
+
+
+@asyncio.coroutine
+def test_new_version_shows_entity_true(hass, mock_get_uuid, mock_get_newest_version):
     """Test if new entity is created if new version is available."""
     mock_get_uuid.return_value = MOCK_HUUID
-    mock_get_newest_version.return_value = mock_coro((NEW_VERSION, ""))
+    mock_get_newest_version.return_value = mock_coro((NEW_VERSION, RELEASE_NOTES))
 
     res = yield from async_setup_component(hass, updater.DOMAIN, {updater.DOMAIN: {}})
     assert res, "Updater failed to set up"
 
+    yield from hass.async_block_till_done()
+
     with patch("homeassistant.components.updater.current_version", MOCK_VERSION):
         async_fire_time_changed(hass, dt_util.utcnow() + timedelta(hours=1))
```

Comment:
Whenever we want to control time in tests we should first patch time so we control exactly what time will be reported.

See eg:
https://github.com/home-assistant/home-assistant/blob/39257164a92e13094ec37da0e1e1b6c50d1bd4c9/tests/components/manual/test_alarm_control_panel.py#L83-L119

Make sure to patch time before the first call to get time happens in the test, eg in our case that's before setting up the updater component.

Reply:
Eg make a pytest fixture that patches utcnow in the correct place. Yield the mock from the fixture. Then we can set the return value for utcnow dynamically in the test as needed.

Reply:
Done. Did I miss anything?